[Map Port] Isla duala

ante185 opened the pull request:
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
Map port project of Isla Duala



### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
Playtesting 
Support in map porting

Requires:
1. [ ] Playtest
2. [ ] Pictures & images
3. [ ] Strings & string tables


********************************************************
Notes:
Posted to be viewable and give space for feedback/help

ante185 commented:
i uh... think i've built all the locations now? which feels a little change?
There is a couple of locations that would be valid or good outpost or resource locations but holding off on them for the moment. easier to add them later than have to take them out.

Feedback on the above would be appreciated, and some response on the map comments, like the ones about potential outpost locations.

ante185 commented:
![image](https://github.com/official-antistasi-community/A3-Antistasi/assets/49555217/e24fd5bd-9277-4ce3-84f4-06fda2796e39)
![image](https://github.com/official-antistasi-community/A3-Antistasi/assets/49555217/057d3a1f-2593-4640-90a8-7327516b159e)

jaj22 commented on A3A/addons/core/Templates/Templates/3CB/3CB_AI_ION_Temperate.sqf:
```diff
@@ -27,7 +27,7 @@
 ["vehiclesMedical", ["UK3CB_ION_B_Desert_Hilux_Ambulance", "UK3CB_ION_B_Desert_M113_AMB", "UK3CB_ION_B_Desert_Transit_Medevac"]] call _fnc_saveToTemplate;
 ["vehiclesLightAPCs", ["UK3CB_B_LAV25_HQ_US_WDL", "UK3CB_ION_B_Desert_M113_M2", "UK3CB_ION_B_Desert_M113_MK19", "UK3CB_KDF_B_MTLB_KPVT", "UK3CB_KRG_B_MTLB_PKT"]] call _fnc_saveToTemplate;
-["vehiclesAPCs", ["UK3CB_B_LAV25_US_WDL", "UK3CB_B_LAV25_US_WDL", "UK3CB_B_AAV_US_WDL", "UK3CB_B_AAV_US_WDL", "RHS_M2A2_wd", "UK3CB_KDF_B_MTLB_BMP", "UK3CB_KDF_B_MTLB_Cannon"]] call _fnc_saveToTemplate;
+["vehiclesAPCs", ["UK3CB_B_LAV25_US_WDL", "UK3CB_B_LAV25_US_WDL", "UK3CB_B_AAV_US_WDL", "UK3CB_B_AAV_US_WDL", "RHS_M2A2_wd", "UK3CB_KDF_B_MTLB_BMP", "UK3CB_KDF_B_MTLB_Cannon",]] call _fnc_saveToTemplate;
```
Needs rebase onto unstable to get rid of these. Guessing that you're working from master rather than unstable locally?

jaj22 commented:
Seems mostly reasonable. Common issues:
- Vehicle markers need a lot more care: The direction of the marker should point in the direction that the spawned vehicles face. They need to be at least 6m wide per vehicle and 10m deep. Outposts should ideally have 3-4 vehicle places. Airports ~8. Other locations can get away with one.
- Some very odd _helipad markers. In general initSpawnPlaces searches for any helipad object (or hangar) within the marker, and so you only need to mark them when outside the marker. If you want to add a helipad within a marker, just add a helipad object.
- Roadblocks and specops markers seem very spammy in places. It will spawn every single one of those. The roadblock autogeneration routine won't put them within 1km of each other, for reference. Note that if you don't have a lot of side-roads, the roadblock autogen might do a decent job anyway, and it's less predictable.
- Some tower placement is a bit spammy. Remember that it'll put a 50cal in every small tower and three in every large tower, plus an AA launcher per HQ building. Will probably adjust that in the future but that's how it goes for now.
- Seaports missing boat spawns. At least I think those were a thing.

ante185 commented:
> *Outposts should ideally have 3-4 vehicle places. Airports ~8. Other locations can get away with one.

That follows the format of outpost_x_vehicle_y right?

> * Roadblocks and specops markers seem very spammy in places. It will spawn every single one of those. The roadblock autogeneration routine won't put them within 1km of each other, for reference. Note that if you don't have a lot of side-roads, the roadblock autogen might do a decent job anyway, and it's less predictable.

Will straight up removing some of them cause any issues, or can the suffix numbers be discontinuous?

> * Some tower placement is a bit spammy. Remember that it'll put a 50cal in every small tower and three in every large tower, plus an AA launcher per HQ building. Will probably adjust that in the future but that's how it goes for now.

I have done a reduction pass prior, about what kinda range of number of statics per outpost are we talking?

> * Seaports missing boat spawns. At least I think those were a thing.

Would that be the seaSpawn marker? not too terribly sure how the three kinds of sea markers should be placed, even after looking at Altis. at least not with the shape of this map

ante185 commented:
I have annotated the map with a couple comments about potential outpost locations - i believe some areas are quite thin and I believe there is room and a need for a couple more outpost.
If i could get some response on those comments/locations I'd be able to get this map to "mostly done" with really only QA and fixes to bring everything up to community standards

jaj22 commented:
Feedback on map notes, going roughly low left to high right:

259353: Yeah should be an outpost.
357308: Could put an outpost on Camp Noobia but I think there's too much stuff in the vicinity already. Ambush Alley is kinda interesting too but roads are iffy.
485363: Small outpost I think. Probably placed with the vehicles down the road from the compound, and knock down a wall so that the AIs don't freak out.
732350: I think the way to do this is to have a fairly thin outpost straddling both the radio tower area and the helipads. Hide the nearest helipad to the road and replace it with vehicle places.
046585: Maybe a bit too useless. Could have a small outpost there I guess, gives people a box to steal.
277473: Road/hill straddle I think: Vehicle places down near the road, some fortifications further up.
348527: Small outpost.
485500: Too much there already I think. Especially once we make it a town.,
518444: Could be a small outpost but maybe too harsh and just a roadblock. Not sure.
583477: No access, no altitude. Leave for specops.
600555: Could be a small outpost. Kinda overkill for the area.
731570: Not a bad location for a small outpost but would need a lot of buildings and the road access is dubious. Not really necessary, so maybe skip it.
843456: Doesn't really do anything on the island, although it's maybe far enough from Canto to be acceptable as an interesting box raid. Should be small though.
395639: Enough here already.
465626: Enough here already.
386733: Obvious resource but probably enough already. Arguable. Being under the palace guns is also... interesting.
556763: Probably outpost just because there's a lot of other stuff to protect within ~2km, although there are also a lot of ouposts. Kinda have to put something there though.
780760: Doesn't cover much and it's a lot of building. Maybe don't bother.
366840: I'd skip that one. Plenty of other outposts on both sides of the bridge.

ante185 commented:
Obmeya radio tower moved to Gerenuk Desert hill outpost.
Added markers for final outposts
Adjusted and supplemented location vehicle markers to minimum values
Removed unneeded/unused helipad markers

ante185 commented:
Now im starting to run out of locations to build or to adjust

ante185 commented:
Rebased.
All locations should be done now, new locations following feedback have been added and built.
It might be ready for review? I'm having a hard time finding anything more to adjust now.

2 things that i want / do know about is 
1. I'd like an review of the navgrid, as i am sure some nodes are messed up from me just not knowing how to set things up.
2. I'm not sure what to do with the sea spawn, sea patrol, and sea attack markers, i've tried to look at Altis for examples but i don't really know how to best place them.

jaj22 commented:
There's a town we can fix with config_fixes now, right?

ante185 commented:
> There's a town we can fix with config_fixes now, right?

I believe that's the rumor, I'd just need an example to follow

jaj22 commented:
I imagine it's just creating a new entry in `Names` for the map:
```
class CfgWorlds {
    class IslaDuala3 {
        class Names {
            class uniqueTownName {
                angle = 0;                          // never seen this non-zero
                name = "ActualVisibleLabel";
                position[] = {5418.33,1502.98};     // 2d map coords
                radiusA = 300;                      // I think these are supposed to be outer/inner radius but Antistasi uses them as X/Y. Probably a bug.
                radiusB = 150;
                type = "NameVillage";               // NameCity also works, I don't think Antistasi cares
            };
        };
    };
};
```
Isla Duala's CfgPatches entry (addon dependency name) is "A3_Map_IslaDuala3" and we can now use `skipWhenMissingDependencies = 1;` in CfgPatches instead of the __has_include stuff.

ante185 commented:
Alright so i think there's just a couple things left to do.

1. Clean up/redo the nav grid, if i've understood correctly a lot of 3 way intersections, especially Y intersections where the roads split off.

2. Go over the outposts that have been added following the last feedback, i think miscommunication and misunderstandings lead to a few more than needed outposts where added. here should be the relevant locations https://github.com/official-antistasi-community/A3-Antistasi/pull/2866#issuecomment-1664855305

3. An overall pass on the outposts (That are left after 2), focused on under-built/over-built locations.

4. Add and modify 1-2 location config following https://github.com/official-antistasi-community/A3-Antistasi/pull/2866#issuecomment-1769109405 as one town needs to be created from scratch and another is set to the wrong type.

1 and 4 are things that i know that i could or can do on my own, however with 2 and 3 i think i definitely need an outside opinion after having stared myself blind at this map.

ante185 commented:
> NameVillage

![image](https://github.com/official-antistasi-community/A3-Antistasi/assets/49555217/419edebf-a1fb-4b0b-84f0-0c5c0af99ffd)
![image](https://github.com/official-antistasi-community/A3-Antistasi/assets/49555217/cbecffef-f813-499d-8f9f-314dc7037be7)
![image](https://github.com/official-antistasi-community/A3-Antistasi/assets/49555217/06558f40-93e3-4160-826d-2a391668c337)

Welp, that could've gone better.
Commented out the offending code for now, but left it in so that someone else could have a look at it.

ante185 commented:
Updated outposts a bit, added some open air plane parking to a couple airbases.
Cleaned up the nav grid as best as i could.

https://github.com/official-antistasi-community/A3-Antistasi/pull/2866#issuecomment-1773966345 is still relevant

jaj22 commented:
Figured out the CfgWorlds/Names issue. IslaDuala3 is actually inherited from Stratis, so you need to write the config like this:
```
class CfgWorlds {
    class Stratis;
    class IslaDuala3 : Stratis {
        class Names {
            // names go here
        };
    };
};
```
Seems to work fine. Radius for Imalia is a bit large.

ante185 commented:
> Figured out the CfgWorlds/Names issue. IslaDuala3 is actually inherited from Stratis, so you need to write the config like this:
> 
> ```
> class CfgWorlds {
>     class Stratis;
>     class IslaDuala3 : Stratis {
>         class Names {
>             // names go here
>         };
>     };
> };
> ```
> 
> Seems to work fine. Radius for Imalia is a bit large.

Thanks, i couldn't find that inheritance when i checked in the config viewer
